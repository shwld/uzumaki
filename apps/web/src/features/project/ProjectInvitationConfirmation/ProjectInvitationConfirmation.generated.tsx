import * as Types from '../../../graphql/generated/graphql';

import gql from 'graphql-tag';
import * as Urql from 'urql';
export type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;
export type ProjectInvitationConfirmation_MemberFragment = {
  __typename?: 'ProjectMember';
  id: string;
  role: Types.ProjectMemberRole;
  name: string;
  avatarImageUrl: string;
};

export type ProjectInvitationConfirmationQueryVariables = Types.Exact<{
  projectId: Types.Scalars['ID'];
  confirmationToken: Types.Scalars['String'];
}>;

export type ProjectInvitationConfirmationQuery = {
  __typename?: 'Query';
  viewer?:
    | {
        __typename?: 'Viewer';
        id: string;
        project?: { __typename?: 'Project'; id: string } | undefined;
        invitationToken?:
          | {
              __typename?: 'ProjectMemberInvitationToken';
              id: string;
              expiredAt: any;
              isExpired: boolean;
              invitation: {
                __typename?: 'ProjectMemberInvitation';
                projectName: string;
              };
            }
          | undefined;
      }
    | undefined;
};

export type ProjectInvitationConfirmation_JoinProjectMemberMutationVariables =
  Types.Exact<{
    input: Types.JoinProjectMemberInput;
  }>;

export type ProjectInvitationConfirmation_JoinProjectMemberMutation = {
  __typename?: 'Mutation';
  joinProjectMember:
    | { __typename?: 'InternalErrorResult' }
    | {
        __typename?: 'InvalidArgumentsResult';
        issues: Array<{
          __typename?: 'ValidationIssue';
          field?: string | undefined;
          message?: string | undefined;
        }>;
      }
    | { __typename?: 'JoinProjectMemberAlreadyJoinedResult'; result: boolean }
    | { __typename?: 'JoinProjectMemberSuccessResult'; result: boolean }
    | {
        __typename?: 'JoinProjectMemberTokenIsAlreadyUsedResult';
        result: boolean;
      }
    | { __typename?: 'JoinProjectMemberTokenIsExpiredResult'; expiredAt: any }
    | { __typename?: 'UnauthorizedResult' };
};

export const ProjectInvitationConfirmation_MemberFragmentDoc = gql`
  fragment ProjectInvitationConfirmation_Member on ProjectMember {
    id
    role
    name
    avatarImageUrl
  }
`;
export const ProjectInvitationConfirmationDocument = gql`
  query ProjectInvitationConfirmation(
    $projectId: ID!
    $confirmationToken: String!
  ) {
    viewer {
      id
      project(id: $projectId) {
        id
      }
      invitationToken(confirmationToken: $confirmationToken) {
        id
        expiredAt
        isExpired
        invitation {
          projectName
        }
      }
    }
  }
`;

export function useProjectInvitationConfirmationQuery(
  options: Omit<
    Urql.UseQueryArgs<ProjectInvitationConfirmationQueryVariables>,
    'query'
  >
) {
  return Urql.useQuery<
    ProjectInvitationConfirmationQuery,
    ProjectInvitationConfirmationQueryVariables
  >({ query: ProjectInvitationConfirmationDocument, ...options });
}
export const ProjectInvitationConfirmation_JoinProjectMemberDocument = gql`
  mutation ProjectInvitationConfirmation_JoinProjectMember(
    $input: JoinProjectMemberInput!
  ) {
    joinProjectMember(input: $input) {
      ... on JoinProjectMemberSuccessResult {
        result
      }
      ... on JoinProjectMemberTokenIsAlreadyUsedResult {
        result
      }
      ... on JoinProjectMemberTokenIsExpiredResult {
        expiredAt
      }
      ... on JoinProjectMemberAlreadyJoinedResult {
        result
      }
      ... on InvalidArgumentsResult {
        issues {
          field
          message
        }
      }
    }
  }
`;

export function useProjectInvitationConfirmation_JoinProjectMemberMutation() {
  return Urql.useMutation<
    ProjectInvitationConfirmation_JoinProjectMemberMutation,
    ProjectInvitationConfirmation_JoinProjectMemberMutationVariables
  >(ProjectInvitationConfirmation_JoinProjectMemberDocument);
}
