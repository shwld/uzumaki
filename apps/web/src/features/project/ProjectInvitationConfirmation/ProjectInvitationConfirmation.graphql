fragment ProjectInvitationConfirmation_Member on ProjectMember {
  id
  role
  name
  avatarImageUrl
}

query ProjectInvitationConfirmation(
  $projectId: ID!
  $confirmationToken: String!
) {
  viewer {
    id
    project(id: $projectId) {
      id
    }
    invitationToken(confirmationToken: $confirmationToken) {
      id
      expiredAt
      isExpired
      invitation {
        projectName
      }
    }
  }
}

mutation ProjectInvitationConfirmation_JoinProjectMember(
  $input: JoinProjectMemberInput!
) {
  joinProjectMember(input: $input) {
    ... on JoinProjectMemberSuccessResult {
      result {
        ...ProjectInvitationConfirmation_Member
      }
    }
    ... on JoinProjectMemberTokenIsAlreadyUsedResult {
      result {
        id
      }
    }
    ... on JoinProjectMemberTokenIsExpiredResult {
      expiredAt
    }
    ... on JoinProjectMemberAlreadyJoinedResult {
      result {
        ...ProjectInvitationConfirmation_Member
      }
    }
    ... on InvalidArgumentsResult {
      issues {
        field
        message
      }
    }
  }
}
