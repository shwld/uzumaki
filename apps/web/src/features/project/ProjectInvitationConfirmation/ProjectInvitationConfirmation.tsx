import { Box, BoxProps, Button, Heading, Text } from '@chakra-ui/react';
import { generateId } from 'core-domain';
import Link from 'next/link';
import { FC } from 'react';
import {
  useProjectInvitationConfirmationQuery,
  useProjectInvitationConfirmation_JoinProjectMemberMutation,
} from './ProjectInvitationConfirmation.generated';

export const ProjectInvitationConfirmation: FC<
  { projectId: string; confirmationToken: string } & BoxProps
> = ({ projectId, confirmationToken, ...props }) => {
  const [{ data, fetching, error }, refetch] =
    useProjectInvitationConfirmationQuery({
      variables: { projectId, confirmationToken },
    });
  const [result, join] =
    useProjectInvitationConfirmation_JoinProjectMemberMutation();

  if (fetching) return <></>;
  if (error != null) return <></>;
  const invitationToken = data?.viewer?.invitationToken;
  if (invitationToken == null) return <></>;

  const canAccessProject = data?.viewer?.project != null;
  return (
    <Box p={5} shadow="md" borderWidth="1px" {...props}>
      <Heading fontSize="xl">
        Invite from {invitationToken.invitation.projectName}
      </Heading>
      {canAccessProject && (
        <>
          <Text>You are already member</Text>
          <Link href={`/projects/${projectId}`}>
            <Text color="green" cursor="pointer">
              GO TO PROJECT
            </Text>
          </Link>
        </>
      )}
      {!canAccessProject && (
        <Button
          colorScheme="green"
          w="full"
          mt="5"
          disabled={result.fetching}
          onClick={async () => {
            await join({
              input: {
                memberId: generateId(),
                confirmationToken,
              },
            });
            refetch();
          }}
        >
          ACCEPT INVITATION
        </Button>
      )}
    </Box>
  );
};
