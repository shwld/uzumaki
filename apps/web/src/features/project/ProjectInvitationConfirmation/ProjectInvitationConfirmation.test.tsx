import { ProjectInvitationConfirmation } from './ProjectInvitationConfirmation';
import { render } from '@testing-library/react';
import { MockedUrqlProvider } from '~/test/MockedUrqlProvider';

describe('ProjectInvitationConfirmation', () => {
  const renderComponent = () => {
    const renderResult = render(
      <MockedUrqlProvider>
        <ProjectInvitationConfirmation
          projectId="projectid"
          confirmationToken="test"
        />
      </MockedUrqlProvider>
    );
    return renderResult;
  };
  test('Snapshot', () => {
    expect(renderComponent().asFragment()).toMatchSnapshot();
  });
  test('success', () => {
    const { getByText } = renderComponent();
    expect(getByText('GO TO PROJECT')).toBeTruthy();
  });
});
