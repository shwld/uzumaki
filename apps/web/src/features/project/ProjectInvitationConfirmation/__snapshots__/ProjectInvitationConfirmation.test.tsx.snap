// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectInvitationConfirmation Snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-1hmx26h"
  >
    <h2
      class="chakra-heading css-6dvxm2"
    >
      Invite from porro
    </h2>
    <p
      class="chakra-text css-0"
    >
      You are already member
    </p>
    <p
      class="chakra-text css-1jilhzb"
    >
      GO TO PROJECT
    </p>
  </div>
</DocumentFragment>
`;
